src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetadataGetIndexInfoTest.java

```diff
@@ -0,0 +1,174 @@
+package com.microsoft.sqlserver.jdbc.databasemetadata;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.jdbc.TestResource;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+
+public class DatabaseMetadataGetIndexInfoTest extends AbstractTest {
+
+	private static String tableName = AbstractSQLGenerator.escapeIdentifier("DBMetadataTestTable");
+    private static String col1Name = AbstractSQLGenerator.escapeIdentifier("p1");
+    private static String col2Name = AbstractSQLGenerator.escapeIdentifier("p2");
+    private static String col3Name = AbstractSQLGenerator.escapeIdentifier( "p3");
+    
+    @BeforeAll
+    public static void setupTests() throws Exception {
+        setConnection();
+    }
+    
+    @BeforeEach
+    public void init() throws Exception {
+        try (Connection con = getConnection()) {
+            con.setAutoCommit(false);
+            try (Statement stmt = con.createStatement()) {
+            	TestUtils.dropTableIfExists(tableName, stmt);
+            	String createTableSQL = "CREATE TABLE " + tableName + " (" +
+                        col1Name + " INT, " +
+                        col2Name + " INT, " +
+                        col3Name + " INT)";
+            	
+            	stmt.executeUpdate(createTableSQL);
+            	assertNull(connection.getWarnings(), "Expecting NO SQLWarnings from 'create table', at Connection.");
+            	assertNull(stmt.getWarnings(), "Expecting NO SQLWarnings from 'create table', at Statement.");

```

Comment:
Because these messages are used several times, can each of the Connection and Statement messages be placed in TestResource instead?